h2o-algos/src/main/java/hex/glm/ComputationState.java

```diff
@@ -874,10 +1048,14 @@ protected double updateState(double [] beta,GLMGradientInfo ginfo){
       if(_beta == null)_beta = beta.clone();
       else System.arraycopy(beta,0,_beta,0,beta.length);
     }
+    if (_parms._linear_constraints != null && _ginfo != null)
+      _relImprovement = (_ginfo._objVal - ginfo._objVal) / Math.abs(_ginfo._objVal);
+    else
+      _relImprovement = (objOld - objective()) / Math.abs(objOld);
```

Comment:
Is there a reason, why with `linear_constrains` you cannot use `objOld` and `objective()`? To me it seems that `ginfo._objVal` should still be the objective (semantically). When adding some other feature, it can get hard to know where to get the true objective so I'd rather stick with methods we already have than adding if branch to every future feature that requires the objective.

Reply:
This is a very good point.  I need to fix it.

Reply:
Fixed.